src/backend/utils/adt/varchar.c

```diff
@@ -518,35 +518,13 @@ varchar_input(const char *s, size_t len, int32 atttypmod)
 		size_t		mbmaxlen = pg_mbcharcliplen(s, len, maxlen);
 		size_t		j;
 
-		if (nls_length_semantics == NLSLENGTH_BYTE)
+		for (j = mbmaxlen; j < len; j++)
```

Comment:
IMO, varchar  also can set unit on modifer. such as varhcar(10 byte) or varchar(10 char). Please confirm it.